### PR TITLE
Support saving search space in experiment load&save command

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: tools/nni_cmd/nnictl_utils.py [hunk 1]
```diff
@@ -929,7 +942,26 @@ def load_experiment(args):
...
 
-    # Step5. Create experiment metadata
+    # Step5. Copy searchSpace file
+    archive_search_space_dir = os.path.join(temp_root_dir, 'searchSpace')
+    target_path = os.path.expanduser(args.searchSpacePath)
+    if not os.path.isabs(target_path):
+        target_path = os.path.join(os.getcwd(), target_path)
+        print_normal('Expand search space path to %s' % target_path)
+    nnictl_exp_config['searchSpacePath'] = target_path
+    # if the path alerady has a search space file, use the original one, otherwise use archived one
```
Comment (ultmaster) [inline tools/nni_cmd/nnictl_utils.py:952]: already

File: tools/nni_cmd/nnictl_utils.py [hunk 2]
```diff
@@ -929,7 +942,26 @@ def load_experiment(args):
...
+    if not os.path.isfile(target_path):
+        if len(os.listdir(archive_search_space_dir)) == 0:
+            print_error('Archive file does not contain search space file!')
+            exit(1)
+        else:
+            for file in os.listdir(archive_search_space_dir):
```
Comment (ultmaster) [inline tools/nni_cmd/nnictl_utils.py:958]: Why do we need a dir?

Comment (SparkSnail) [reply tools/nni_cmd/nnictl_utils.py:958]: For storage data and classify. There are a few folders in archive folder, including 'code', 'nnictl', 'experiment', 'searchSpace', the search space file's name is random, put this file into 'searchSpace' folder for identify.

File: tools/nni_cmd/nnictl_utils.py (hunk 1 above)
Comment (SparkSnail) [reply tools/nni_cmd/nnictl_utils.py]: updated.

File: tools/nni_cmd/nnictl_utils.py [hunk 3]
```diff
@@ -929,7 +942,26 @@ def load_experiment(args):
...
+                source_path = os.path.join(archive_search_space_dir, file)
+                os.makedirs(os.path.dirname(target_path), exist_ok=True)
+                shutil.copyfile(source_path, target_path)
+                break
```
Comment (ultmaster) [inline tools/nni_cmd/nnictl_utils.py:962]: Throw error when no file is found?

Comment (SparkSnail) [reply tools/nni_cmd/nnictl_utils.py:962]: there is already a judgement in lint 955, don't need to throw error here anymore.

File: tools/nni_cmd/nnictl.py [hunk 4]
```diff
@@ -159,6 +159,8 @@ def parse_args():
     parser_load_experiment.add_argument('--codeDir', '-c', required=True, help='the path of codeDir for loaded experiment, \
                                    this path will also put the code in the loaded experiment package')
     parser_load_experiment.add_argument('--logDir', '-l', required=False, help='the path of logDir for loaded experiment')
+    parser_load_experiment.add_argument('--searchSpacePath', '-s', required=True, help='the path of search space file for \
```
Comment (chicm-ms) [inline tools/nni_cmd/nnictl.py:162]: Some experiments do not have search space, this should be an optional parameter.

**Review by ultmaster — APPROVED** [review]

File: tools/nni_cmd/nnictl.py (hunk 4 above)
Comment (SparkSnail) [reply tools/nni_cmd/nnictl.py]: updated.

File: tools/nni_cmd/nnictl_utils.py [hunk 5]
```diff
@@ -827,7 +827,18 @@ def save_experiment(args):
         temp_code_dir = os.path.join(temp_root_dir, 'code')
         shutil.copytree(nni_config.get_config('experimentConfig')['trial']['codeDir'], temp_code_dir)
 
-    # Step4. Archive folder
+    # Step4. Copy searchSpace file
+    search_space_path = nni_config.get_config('experimentConfig').get('searchSpacePath')
+    if search_space_path:
+        if not os.path.exists(search_space_path):
+            print_error('search space %s does not exist!' % search_space_path)
```
Comment (QuanluZhang) [inline tools/nni_cmd/nnictl_utils.py:834]: warning might be better?

Comment (SparkSnail) [reply tools/nni_cmd/nnictl_utils.py]: updated.

**Review by QuanluZhang — APPROVED** [review]

**Review by chicm-ms — APPROVED** [review]

**Pull request merged**